src/test/java/org/jenkinsci/plugins/gitclient/GitAPITestCase.java

```diff
@@ -865,7 +865,14 @@ public void test_clean() throws Exception {
          */
         String fileName = "\uD835\uDD65-\u5c4f\u5e55\u622a\u56fe-\u0041\u030a-\u00c5-\u212b-fileName.xml";
         w.touch(fileName, "content " + fileName);
-        w.git.add(fileName);
+        try {
```

Comment:
The problem with many tests in this project is, that in a single test method many assertions for different cases are performed. This is the case here too.

This improved error reporting should be performed for all assertions in this test method - lines [869](https://github.com/MarkEWaite/git-client-plugin/blob/de9da9a7f942e113b919b200b1811332acdb6aeb/src/test/java/org/jenkinsci/plugins/gitclient/GitAPITestCase.java#L869), [883](https://github.com/MarkEWaite/git-client-plugin/blob/de9da9a7f942e113b919b200b1811332acdb6aeb/src/test/java/org/jenkinsci/plugins/gitclient/GitAPITestCase.java#L883), [888](https://github.com/MarkEWaite/git-client-plugin/blob/de9da9a7f942e113b919b200b1811332acdb6aeb/src/test/java/org/jenkinsci/plugins/gitclient/GitAPITestCase.java#L888).

I suggest to implement a helper method which wraps the improved reporting. With the help of Java 8 lambdas this even reads quite well.

line 869
```
withSystemLocaleReporting(fileName, () -> {
    w.git.add(fileName);
    w.git.commit(fileName);
});
```

line 883
```
withSystemLocaleReporting(fileNameSwim, () -> {
    w.git.add(fileNameSwim);
    w.git.commit(fileNameSwim);
});
```

line 888
```
withSystemLocaleReporting(fileNameFace, () -> {
    w.git.add(fileNameFace);
    w.git.commit(fileNameFace);
});
```

helper method
```
private void withSystemLocaleReporting(String fileName, TestedCode code) throws Exception {
    try {
        code.run();
    } catch (GitException ge) {
        // Exception message should contain the actual file name.
        // It may just contain ? for characters that are not encoded correctly due to the system locale.
        // If such a mangled file name is seen instead, throw a clear exception to indicate the root cause.
        assertTrue("System locale does not support filename '" + fileName + "'", ge.getMessage().contains("?"));
        // Rethrow exception for all other issues.
        throw ge;
    }
}

@FunctionalInterface
interface TestedCode {
    void run() throws Exception;
}
```